### PR TITLE
More porting Microsoft.Build to .NET Core

### PR DESCRIPTION
**Pull request by dsplaisted**
More progress towards getting Microsoft.Build to compile for .NET Core

### CODE REVIEW

File: src/Shared/FileUtilities.GetFolderPath.cs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+using System.Text.RegularExpressions;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Build.Shared
+{
+    internal static partial class FileUtilities
+    {
+        //  .NET Core doesn't currently have APIs to get the "special" folders (ie the ones defined in the Environment.SpecialFolder enum)
+        //  The code here is mostly copied out of the .NET reference source for this functionality
```
Comment (akoeplinger) [inline src/Shared/FileUtilities.GetFolderPath.cs]: Are you taking stuff from https://github.com/Microsoft/referencesource or http://referencesource.microsoft.com? The latter isn't MIT licensed.

Comment (dsplaisted) [inline src/Shared/FileUtilities.GetFolderPath.cs]: @akoeplinger I'm doing this work as a Microsoft employee.  Since we own the code, we'

Comment (dsplaisted) [inline src/Shared/FileUtilities.GetFolderPath.cs]: @akoeplinger Excellent point!

This PR is provided as part of my job at Microsoft, who is the copyright holder on the code. By providing this PR, we're licensing the code submitted in this PR under the license of this project, which is MIT. 

In this case, the code copied from the reference source was already part of the MIT-licensed reference source at GitHub.

Comment (akoeplinger) [inline src/Shared/FileUtilities.GetFolderPath.cs]: Thanks for clarifying :+1: I know you're an MS employee so I just wanted to have something on record that clarifies this, licensing is hard :smile:

**Comment by rainersigwald** [comment]
LGTM.

**Pull request merged**